### ISSUE TITLE
Issue plotting big endian images

### ISSUE THREAD

**Issue by jenshnielsen** [matplotlib/matplotlib#6671]
Originally reported by Anik Halder on the mailing list http://matplotlib.1069221.n5.nabble.com/IPython-FITS-file-plotting-problem-tp47271.html

Consider this simple example:

```python
import matplotlib.pyplot as plt
import numpy as np

x = np.arange(10)
X, Y = np.meshgrid(x,x)
Z = ((X-5)**2 + (Y-5)**2)**0.5

plt.imshow(Z, origin = "lower")
plt.colorbar()
plt.show()

plt.figure()
plt.imshow(Z.astype('>f8'), origin = "lower")
plt.colorbar()
plt.show()
```
Which produces:
![right](https://cloud.githubusercontent.com/assets/548266/16527177/3c6c6a86-3fad-11e6-843e-6a17a6afeab3.png)
And:
![wrong](https://cloud.githubusercontent.com/assets/548266/16527178/3e7af0cc-3fad-11e6-9b95-fdd75809b79e.png)

This works correctly in 1.5.1 so it looks like it's a regression in 2.x

**Comment by tacaswell** [comment]
I think this is a duplicate of https://github.com/matplotlib/matplotlib/issues/6394 which was fixed in https://github.com/matplotlib/matplotlib/pull/6422

**Comment by jenshnielsen** [comment]
It sounds like that should have fixed it however I can still reproduce it with 2.0.0b1+98.gc4fa469 which I just installed.

**Comment by QuLogic** [comment]
It works on `v1.5.x`, but not `v2.x`. Bisect points to commit 1302606cbfe03ca8d00eb80f99b33e1b11091b65
```
Author: Thomas A Caswell <tcaswell@gmail.com>
Date:   Fri Mar 4 00:17:47 2016 -0500

    Merge pull request #6099 from mdboom/masked-image
    
    Fix #6069.  Handle image masks correctly

:040000 040000 7daaffa94619d32e28aa44502209445b5670cc8b c5e94ccc04761d2b02be7a55bd07598dc4830236 M	lib
:040000 040000 98a3d3dc7f7a61d7029d97f9f5e657f77cb582ec 6b646c30374dc76196d8b2b0a406d06e7f2159e2 M	src
```

**Comment by QuLogic** [comment]
More specifically:

```
eda09124a3d262bb47405924f44cdf11853f2b1e is the first bad commit
commit eda09124a3d262bb47405924f44cdf11853f2b1e
Author: Michael Droettboom <mdboom@gmail.com>
Date:   Wed Mar 2 10:46:46 2016 -0500

    Fix #6069.  Handle image masks correctly
    
    Alternative to #6070
    
    This works by converting the greyscale image to an RGBA image (with the
    alpha channel created correctly) prior to resizing in Agg.  This also
    removes the hack (that caused problems with over/under) where we were
    using negative values to indicate transparency.

:040000 040000 7daaffa94619d32e28aa44502209445b5670cc8b ff69631973998fa2efe6a5f932b9a79f02fdd088 M	lib
```

Also, edited to post to include creation of `Z` from the notebook.

**Comment by tacaswell** [comment]
Ok, this behavior makes sense.

```python
In [13]: matplotlib.cbook.safe_masked_invalid(Z.astype('>f8'), copy=True).data.dtype
Out[13]: dtype('>f8')
```

numpy is happily copy, but not forcing the data to be native.  I have a PR on the way.
